aiven/resource_account_authentication.go

```diff
@@ -0,0 +1,225 @@
+package aiven
+
+import (
+	"github.com/aiven/aiven-go-client"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+var aivenAccountAuthenticationSchema = map[string]*schema.Schema{
+	"account_id": {
+		Type:        schema.TypeString,
+		Description: "Account id",
+		Required:    true,
+	},
+	"type": {
+		Type:         schema.TypeString,
+		Description:  "Account authentication id",
+		Required:     true,
+		ValidateFunc: validation.StringInSlice([]string{"internal", "saml"}, false),
```

Comment:
What happens when we add the next authentication type? Does this fail when response (authentication methods list) contains entries with something like `openid`? We are not currently planning on adding OpenID support, but I'm fairly certain we'll be adding new authentication method types, and old official clients and libraries shouldn't break when that happens (if user creates a new method directly through aiven console).

Reply:
When we add new authentication type, we have first to update go-client and add all corresponding fields, and then TF provider. I believe if we add a new authentication type on API side, it won't work out of the box in Terraform, and it will require new code to be added in both go-client and TF provider and tested.